### PR TITLE
Update mocha & nyc version

### PR DESCRIPTION
**Pull request by saemjeon**
Update Mocha  & nyc version
mocha: 5.2.0 -> 8.3.2
@types/mocha: 5.2.5 -> 8.2.2
nyc: 14.0.0 -> 15.1.

### CODE REVIEW

**Review by pmconne — COMMENTED** [review]
Looks fine but I defer to someone like @wgoehrig.

**Review by grigasp — APPROVED** [review]

**Review by Animagne — APPROVED** [review]

**Review by GerardasB — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Comment by wgoehrig** [comment]
@grigasp the presentation coverage requirements are failing after updating nyc - can you take a look at those?

**Comment by grigasp** [comment]
> @grigasp the presentation coverage requirements are failing after updating nyc - can you take a look at those?

done

**Comment by calebmshafer** [comment]
@saemjeon can you do the merge required here?  I'll override the massive number of reviewers here to get it merged.

**Comment by saemjeon** [comment]
@calebmshafer I think I did it. Please let me know if I didn't do it correctly.

File: presentation/backend/src/presentation-backend.ts [hunk 1]
```diff
@@ -8,14 +8,11 @@
...
-export { Presentation, PresentationProps, PresentationPropsDeprecated, PresentationPropsNew } from "./presentation-backend/Presentation"; // eslint-disable-line deprecation/deprecation
-export {
-  HierarchyCacheMode, HierarchyCacheConfig, HierarchyCacheConfigBase, HybridCacheConfig, MemoryHierarchyCacheConfig,
-  DiskHierarchyCacheConfig, PresentationManager, PresentationManagerProps, PresentationManagerMode,
-} from "./presentation-backend/PresentationManager";
-export { RulesetManager } from "./presentation-backend/RulesetManager";
-export { RulesetVariablesManager } from "./presentation-backend/RulesetVariablesManager";
-export { RulesetEmbedder, RulesetEmbedderProps, DuplicateRulesetHandlingStrategy } from "./presentation-backend/RulesetEmbedder";
+export * from "./presentation-backend/Presentation";
+export * from "./presentation-backend/PresentationManager";
```
Comment (calebmshafer) [inline presentation/backend/src/presentation-backend.ts:12]: @grigasp you good with this change?

File: tools/build/scripts/test-tsnode.js [hunk 2]
```diff
@@ -1,67 +1,67 @@
...
+
+const debugOptions = argv.debug || argv.inspect ? [
+  "--inspect=9229"
+] : [];
+
+let grepOptions = [];
+if (argv.grep) {
+  grepOptions = ["--grep", argv.grep];
+  if (argv.invert) {
+    grepOptions.push("--invert");
```
Comment (calebmshafer) [inline tools/build/scripts/test-tsnode.js:52]: Can you revert this file? I think it changed the line endings.

Comment (saemjeon) [reply tools/build/scripts/test-tsnode.js]: Fixed it.

**Review by calebmshafer — APPROVED** [review]
LGTM. If Grigas approves of the presentation changes, I'll override the rest of the reviewers tomorrow.

File: presentation/backend/src/presentation-backend.ts (hunk 1 above)
Comment (grigasp) [reply presentation/backend/src/presentation-backend.ts:12]: Yes, I was going to make this change at some point anyway. Thanks.

**Comment by calebmshafer** [comment]
@saemjeon can you take a look at the failure in the integration tests? https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=877365&view=logs&j=11a1ff10-dba1-50ce-27e2-db260e9900bf&t=575d2bbe-c3a7-5f15-a080-f6c1d4bf0611

It looks like there is some issue with using `this.skip` in the Rpc Interface tests.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**